crates/core/src/ring/connection_manager.rs

```diff
@@ -45,6 +45,7 @@ impl ConnectionManager {
         let loc = conn.location.location.unwrap_or_else(Location::random);
         let mut conns = self.connections_by_location.write();
         conns.entry(loc).or_default().push(conn);
+        self.open_connections.fetch_add(1, std::sync::atomic::Ordering::SeqCst);
```

Comment:
this is weird; we are doing a fetch_add(1) in `ring.rs:421` are those conflicting? we may be countign the same connection twice after this change, ensure this is correct

Reply:
I guess we need to clean up this a bit so this function is called by Ring::add_connection and then we can remove the fetch_add from there.